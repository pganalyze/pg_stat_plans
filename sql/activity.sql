
SELECT pg_stat_plans_reset() IS NOT NULL AS t;

--
-- check if we see our own plan in activity
--

SELECT plan FROM pg_stat_plans_activity WHERE pid = pg_backend_pid();


--
-- check if we handle showing our plan for named prepared statements correctly
--

PREPARE x AS SELECT plan FROM pg_stat_plans_activity WHERE pid = pg_backend_pid();
EXECUTE x;
DEALLOCATE x;

SELECT pg_stat_plans_reset() IS NOT NULL AS t;
