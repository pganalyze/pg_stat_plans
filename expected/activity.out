SELECT pg_stat_plans_reset() IS NOT NULL AS t;
 t 
---
 t
(1 row)

--
-- check if we see our own plan in activity
--
SELECT plan FROM pg_stat_plans_activity WHERE pid = pg_backend_pid();
                                                         plan                                                          
-----------------------------------------------------------------------------------------------------------------------
 Hash Join                                                                                                            +
   Hash Cond: ((p.dbid = a.datid) AND (p.userid = a.usesysid) AND (p.queryid = a.query_id) AND (p.planid = a.plan_id))+
   ->  Function Scan on pg_stat_plans p                                                                               +
         Filter: (toplevel IS TRUE)                                                                                   +
   ->  Hash                                                                                                           +
         ->  Function Scan on pg_stat_plans_get_activity a                                                            +
               Filter: (pid = pg_backend_pid())
(1 row)

--
-- check if we handle showing our plan for named prepared statements correctly
--
PREPARE x AS SELECT plan FROM pg_stat_plans_activity WHERE pid = pg_backend_pid();
EXECUTE x;
                                                         plan                                                          
-----------------------------------------------------------------------------------------------------------------------
 Hash Join                                                                                                            +
   Hash Cond: ((p.dbid = a.datid) AND (p.userid = a.usesysid) AND (p.queryid = a.query_id) AND (p.planid = a.plan_id))+
   ->  Function Scan on pg_stat_plans p                                                                               +
         Filter: (toplevel IS TRUE)                                                                                   +
   ->  Hash                                                                                                           +
         ->  Function Scan on pg_stat_plans_get_activity a                                                            +
               Filter: (pid = pg_backend_pid())
(1 row)

DEALLOCATE x;
SELECT pg_stat_plans_reset() IS NOT NULL AS t;
 t 
---
 t
(1 row)

